src/towncrier/test/test_check.py

```diff
@@ -148,3 +172,52 @@ def test_none_stdout_encoding_works(self):
 
         self.assertEqual(0, proc.returncode)
         self.assertEqual(b"", stderr)
+
+    def test_first_release(self):
+        runner = CliRunner()
+
+        with runner.isolated_filesystem():
+            # Arrange
+            write("towncrier.toml", "[tool.towncrier]")
+            write("newsfragments/.gitignore", "!.gitignore")
+            initial_commit()
```

Comment:
This looks like duplicate code between tests.
Why not reuse the existing `create_project`


```suggestion
            create_project('towncrier.toml')
```

Reply:
I dislike `create_project()`.
* It encapsulates a poorly scoped amount of code. The semantics are unclear. It does too many unrelated things. Changing it like you did in #296 affects every test that uses it which is risky.
* It is inflexible: you either use it or you don't. I prefer my approach: a set of (mostly) orthogonal building blocks that every test can use to arrange things precisely the way that particular test needs to.
* It is a little to implicit for my taste. I actually like that every file the test creates is explicitly named in the test itself.

Reply:
> Changing it like you did in #296 affects every test that uses it which is risky.

Just to be clear: I'm not criticizing you, I'm criticizing a specific decision you made because I think I know how to solve that specific problem better.

Reply:
@hexagonrecursion  don't worry :) If you think that some code is not ok. please say it loud.
I will not take it as an offense... I know that I am capable of writing bad code :)

Your comments are valid.

Regarding the risky change. As long as the tests are not failing, I think that any change should be done if we thing that it improves the readability and usability.
We should be confident that the test suite is solid and don't code with fear :)

If you think that `create_project` is bad pattern, it's best to add a docstring comment to it and let other developers know what is the current best practice.

But I think that all test helpers should have short docstrings describing when and why they should be used :)

Reply:
> I know that I am capable of writing bad code

Same here. I also prefer to err on the side of not offending anybody.

> If you think that create_project is bad pattern, it's best to add a docstring comment to it and let other developers know what is the current best practice.

If you are onboard with my argument I'll submit a followup PR to refactor the rest of the file and delete `create_project()`

> But I think that all test helpers should have short docstrings describing when and why they should be used :)

:+1: